src/main/java/org/neo4j/junit/jupiter/causal_cluster/Neo4jServer.java

```diff
@@ -30,6 +31,15 @@
  */
 public interface Neo4jServer {
 
+	/**
+	 * Server can either be a core or replica server.
+	 */
+	enum Type {
+		CORE_SERVER,
+		REPLICA_SERVER,
+		UNKNOWN
```

Comment:
UNKNOWN feels like a bit of a code smell that we could do without. But it's not a big deal.